### PR TITLE
fix: (mono) reclaim entity id after failed auto-create; avoid `FAIL_INVALID` on invalid token id

### LINKED ISSUE #11876: Expected 'TOKEN_NOT_ASSOCIATED_TO_ACCOUNT', got 'SPENDER_DOES_NOT_HAVE_ALLOWANCE'
### Description

Hashscan transaction [here](https://hashscan.io/mainnet/transaction/1708329881.355363003)

Transaction body:
```
transactionID {
  transactionValidStart {
    seconds: 1708329870
    nanos: 360112695
  }
  accountID {
    accountNum: 1104922
  }
}
nodeAccountID {
  accountNum: 29
}
transactionFee: 100000000
transactionValidDuration {
  seconds: 120
}
memo: "Swapping NFTs 0 / 1 - Altlantis"
cryptoTransfer {
  transfers {
  }
  tokenTransfers {
    token {
      tokenNum: 4110805
    }
    nftTransfers {
      senderAccountID {
        accountNum: 692756
      }
      receiverAccountID {
        accountNum: 995022
      }
      serialNumber: 359
      is_approval: true
    }
  }
}
```

### Steps to reproduce

Re-run events from round 162910486

### Additional context

Probably related to #11849 

### Hedera network

other

### Version

11852-pay-rewards-in-triggered-txn

### Operating system

macOS

### LINKED ISSUE #12246: (3/12 03:30) Mismatched values, expected '4931514', got '4931513' - Matching field 'accountNum' Matching field 'accountID' Matching field 'receipt'
### Description

Hashscan [tx](https://hashscan.io/mainnet/transaction/1710214749.019358395):

```
---- CryptoCreate DIFFERED (TRANSACTION_RECORD_MISMATCH) ----
Mismatched values, expected '4931514', got '4931513' - Matching field 'accountNum' Matching field 'accountID' Matching field 'receipt'  ==> expected: <4931514> but was: <4931513>

For body,
transactionID {
  transactionValidStart {
    seconds: 1710214735
    nanos: 432569891
  }
  accountID {
    accountNum: 4614889
  }
}
nodeAccountID {
  accountNum: 30
}
transactionFee: 10000000000
transactionValidDuration {
  seconds: 120
}
cryptoCreateAccount {
  key {
    ed25519: "Jc\347\266\240\313\310\275H\226\357\231L:\301u#\020\2179\256yn\340\235&\354\350ns\364\377"
  }
  sendRecordThreshold: 9223372036854775807
  receiveRecordThreshold: 9223372036854775807
  autoRenewPeriod {
    seconds: 7776000
  }
}
➡️  Expected Record receipt {
  status: SUCCESS
```
These streams differed 385 times.

### Steps to reproduce

Run 15 mins from round 165012101.

### Additional context

_No response_

### Hedera network

other

### Version

release/0.48

### Operating system

None

### PR DESCRIPTION
**Pull request by tinker-michaelj**
**Description**:
 - Closes #12246 
 - Closes #11876
 - If a `TransferLogic` auto-creation fails, reset the next entity id to what it was before attempting the transfer.
 - Ensure `CryptoApproveAllowance` validation returns the failure reason for a thrown `HandleException` instead of propagating it.
 - Fixes 3 `HapiSpec` expectations.

### CODE REVIEW

**Review by iwsimon — DISMISSED** [review]
LGTM. Thanks @tinker-michaelj

**Review by mhess-swl — DISMISSED** [review]
Looks good from services 👍

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/transfer/TransferContextImpl.java [hunk 1]
```diff
@@ -186,6 +190,21 @@ public void validateHbarAllowances() {
                         accountStore.get(aa.accountIDOrElse(AccountID.DEFAULT)), topLevelPayer, aa.amount());
             }
         }
+        final var nftStore = context.readableStore(ReadableNftStore.class);
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/transfer/TransferContextImpl.java:193]: Why do we need to validate only Hbar and NFT allowances here (not token transfers)🤔 ?

**Review by iwsimon — DISMISSED** [review]
LGTM

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/transfer/TransferContextImpl.java (hunk 1 above)
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/transfer/TransferContextImpl.java:193]: Confirmed that these should be deleted after differential testing. It is just to match the mono-service responses. 

A FUTURE removal comment will be added in later PR.

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM ! Thanks @tinker-michaelj

**Review by iwsimon — APPROVED** [review]

**Review by mhess-swl — APPROVED** [review]

**Pull request merged**